engine/src/main/java/org/camunda/bpm/engine/impl/cmd/SetJobRetriesCmd.java

```diff
@@ -16,54 +16,108 @@
  */
 package org.camunda.bpm.engine.impl.cmd;
 
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
 import org.camunda.bpm.engine.ProcessEngineException;
+import org.camunda.bpm.engine.history.UserOperationLogEntry;
 import org.camunda.bpm.engine.impl.cfg.CommandChecker;
 import org.camunda.bpm.engine.impl.interceptor.Command;
 import org.camunda.bpm.engine.impl.interceptor.CommandContext;
 import org.camunda.bpm.engine.impl.persistence.entity.JobDefinitionEntity;
 import org.camunda.bpm.engine.impl.persistence.entity.JobDefinitionManager;
+import org.camunda.bpm.engine.impl.persistence.entity.JobEntity;
 import org.camunda.bpm.engine.impl.persistence.entity.PropertyChange;
-
-import java.io.Serializable;
+import org.camunda.bpm.engine.impl.util.ClockUtil;
 
 
 /**
  * @author Askar Akhmerov
  */
-public class SetJobRetriesCmd extends AbstractSetJobRetriesCmd implements Command<Void>, Serializable {
+public class SetJobRetriesCmd implements Command<Void>, Serializable {
 
   protected static final long serialVersionUID = 1L;
 
+  protected static final String RETRIES = "retries";
+  protected static final String DUE_DATE = "dueDate";
 
   protected final String jobId;
   protected final String jobDefinitionId;
+  protected final List<String> jobIds;
   protected final int retries;
+  protected Date dueDate;
+  protected final boolean isDueDateSet;
 
-
-  public SetJobRetriesCmd(String jobId, String jobDefinitionId, int retries) {
-    if ((jobId == null || jobId.isEmpty()) && (jobDefinitionId == null || jobDefinitionId.isEmpty())) {
-      throw new ProcessEngineException("Either job definition id or job id has to be provided as parameter.");
-    }
-
-    if (retries < 0) {
-      throw new ProcessEngineException("The number of job retries must be a non-negative Integer, but '" + retries + "' has been provided.");
-    }
-
+  public SetJobRetriesCmd(String jobId, String jobDefinitionId, int retries, Date dueDate, boolean isDueDateSet) {
     this.jobId = jobId;
     this.jobDefinitionId = jobDefinitionId;
+    this.jobIds = null;
     this.retries = retries;
+    this.dueDate = dueDate;
+    this.isDueDateSet = isDueDateSet;
+  }
+
+  public SetJobRetriesCmd(List<String> jobIds, int retries, Date dueDate, boolean isDueDateSet) {
+    this.jobId = null;
+    this.jobDefinitionId = null;
+    this.jobIds = jobIds;
+    this.retries = retries;
+    this.dueDate = dueDate;
+    this.isDueDateSet = isDueDateSet;
   }
 
   public Void execute(CommandContext commandContext) {
+    if(dueDate == null && commandContext.getProcessEngineConfiguration().isEnsureJobDueDateNotNull()) {
+      dueDate = ClockUtil.getCurrentTime();
+    }
     if (jobId != null) {
-      setJobRetriesByJobId(jobId, retries, commandContext);
-    } else {
+      setJobRetriesByJobId(jobId, commandContext);
```

Comment:
We can not simply remove it because the method is not always called with the class-level property. Of course, we could refactor the second call (which iterates over the list of ids and calls `setJobRetriesByJobId` for each id in the list) but I think the current solution is the best solution.